module/x/gravity/keeper/keeper.go

```diff
@@ -513,3 +513,53 @@ func (k Keeper) setLastObservedSignerSetTx(ctx sdk.Context, signerSet types.Sign
 	key := []byte{types.LastObservedSignerSetKey}
 	ctx.KVStore(k.storeKey).Set(key, k.cdc.MustMarshal(&signerSet))
 }
+
+// CreateContractCallTx xxx
+func (k Keeper) CreateContractCallTx(ctx sdk.Context, invalidationNonce uint64, invalidationScope tmbytes.HexBytes,
+	payload []byte, tokens []types.ERC20Token, fees []types.ERC20Token) *types.ContractCallTx {
+	params := k.GetParams(ctx)
+
+	newContractCallTx := &types.ContractCallTx{
+		InvalidationNonce: invalidationNonce,
+		InvalidationScope: invalidationScope,
+		Address:           k.getBridgeContractAddress(ctx),
+		Payload:           payload,
+		Timeout:           params.TargetEthTxTimeout,
+		Tokens:            tokens,
+		Fees:              fees,
```

Comment:
Big Q here is how to pay these fees cc @zmanian

Reply:
This is the biggest Q